eclipse-collections-code-generator/src/main/resources/test/set/immutable/abstractImmutablePrimitiveSetTestCase.stg

```diff
@@ -66,12 +69,29 @@ public abstract class AbstractImmutable<name>HashSetTestCase extends AbstractImm
     {
         <name>ArrayList collisions = new <name>ArrayList();
         <name>HashSet set = new <name>HashSet();
-        for (<type> i = <(literal.(type))("32")>; collisions.size() \<= 10; i++)
+
+        Method m;
+        try
+        {
+            m = set.getClass().getDeclaredMethod("spreadAndMask", <type>.class);
```

Comment:
This is a bit of a funny one because this method is accessible only form the mutable packages. After the move to immutable, this is no longer available. I thought of a few options to address that:

1) Expose to public interface but I doubt this is what we want to do
2) Pull all these helper methods into a util class that's written in Java and imported in all tests, accessible by both mutable & immutable packages
3) Do the same as 2 but thru the template generation
4) Use reflection which isn't great but seems like the least impactful change and is test code.. that's why I chose this one

Reply:
Thanks for the clarification @Desislav-Petrov and providing some alternatives. I'm looking at the source and will see if I can find any other alternatives.

Reply:
Ok, taking a look at `AbstractIntSetTestCase` I can see there is a duplicate protected method named `generateCollisions1`. This is in the mutable package. 

Here's what I would recommend. Make the method `generateCollisions1` public, and change the generateCollisions implementation to call the static method on `AbstractIntSetTestCase`. After the change the method should look like this on the abstract immutable test case.

    protected static IntArrayList generateCollisions()
    {
        return AbstractIntSetTestCase.generateCollisions1();
    }

This creates a dependency between the immutable package on the immutable, but that dependency already existed. This is all test code anyway, so I would rather remove duplication than worry that a static test method is protected or public.

Reply:
Ah - haven't thought about that - thx for pointing out, it does seem like a cleaner approach. Changes made to reflect it